Create helper function for RPC profiling in _invoke_rpc and remote

lequytra opened the pull request:
Summary:
Refactor enabling rpc profiling logic in `_invoke_rpc` and `remote()` into `_rpc_profiling()` helper function.

T89058933

Test Plan: Not tested.

Differential Revision: D27922286

rohan-varma commented on torch/distributed/rpc/api.py:
```diff
@@ -640,26 +621,7 @@ def _invoke_rpc(to, func, rpc_type, args=None, kwargs=None, rpc_timeout=UNSET_RP
-        RemoteProfilerManager.set_current_profiling_key(rpc_profiling_key)
-        # Mypy doesn't support re-def of a variable not in the same block (#1174)
-        ctx_manager = torch.autograd.profiler.record_function(rpc_profiling_key)  # type: ignore[assignment]
+        ctx_manager = _enable_rpc_profiler(qualified_name, func, rpc_type, dst_worker_info)
```
Could you also include the logic for `should_profile` and contextlib.suppress() to the helper function (i.e. the code starting on 620)? Also, feel free to remove the TODO above

rohan-varma reviewed (commented):
Awesome, looks great! Just one minor comment and should be good to go

Commit:
Create helper function for RPC profiling in _invoke_rpc and remote (#56643)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/56643

Refactor enabling rpc profiling logic in `_invoke_rpc` and `remote()` into `_rpc_profiling()` helper function.

Differential Revision: D27922286

fbshipit-source-id: 817d69439c6419fbbe7f901624a40aae1bc5519e

rohan-varma reviewed (approved):
Awesome, looks good! please wait for all ci to pass before landing
